### ISSUE TITLE
Font corruption after resume from suspend/hibernate

### ISSUE THREAD

**Issue by raqua** [linuxmint/Cinnamon#1310]
On my Dell Precission 4700, when I resume from suspend or hibernate, I get corrupted fonts in cinnamon. 
See picture here: http://minus.com/l2iwej4Qq4gN6

This does not happen in pure Gnome. 

Reloading Cinnamon solves the problem.

**Comment by jrial** [comment]
Still getting this on Mint 17.2, Cinnamon 2.6.13+rafaela and restarting Cinnamon with alt-f2, r doesn't help.
Intel drivers, in case that matters, but I believe it did the same with the NVidia drivers.

The corruption is even visible in the login screen after resuming.

**Comment by alex-t-grecu** [comment]
Still running into such issue when resuming from Sleep in 
"Linux Mint 18.1 Serena"
kernel 4.4.0-66-generic #87-Ubuntu SMP Fri Mar 3 15:29:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Cinnamon 3.2.7
It seems I was able to recover the correct font display after trying most of the solutions on the web:
- warm restart of Cinnamon
- playing around with xrandr enabling/disabling displays
- changing the default font in the "Fonts" configure tool - that helped partially!
- sudo fc-cache -f -v
- sudo fc-cache -r -v
- and most importantly sudo swapoff -a
I suspect some of the cached fonts got moved from memory to swap before the computer went into sleep mode and the cache was corrupted on resume and actually the last action had the best effect.
I hope this will help developers solve the issue as it seems to be mostly related to the interaction between power modes and low physical memory status. Maybe some buffers should be forced not to ever be swapped.
